### PR TITLE
[CI CD] added direct maven central publish

### PR DESCRIPTION
**Pull request by YannickWeber**
**Changes**
* Adapted build script to directly publish to maven central

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -271,44 +272,29 @@ allprojects {
...
-            publish = true
-            pkg.apply {
-                userOrg = "hivemq"
-                repo = "HiveMQ"
-                name = "hivemq-mqtt-client"
-                desc = project.description
-                websiteUrl = metadata.url.get()
-                issueTrackerUrl = metadata.issueManagement!!.url.get()
-                vcsUrl = metadata.scm!!.url.get()
-                setLicenses(metadata.license!!.shortName.get())
```
Comment (DC2-DanielKrueger) [inline build.gradle.kts:286]: is the license handling missing?

Comment (YannickWeber) [reply build.gradle.kts:286]: No, this is done by the metadata plugin.

**Review by DC2-DanielKrueger — APPROVED** [review]

File: build.gradle.kts [hunk 2]
```diff
@@ -269,46 +270,29 @@ allprojects {
...
-                version.apply {
-                    released = Date().toString()
-                    vcsTag = "v${project.version}"
-                    gpg.apply {
-                        sign = true
+        afterEvaluate {
+            publishing {
+                publications {
+                    create<MavenPublication>("sonatype") {
+                        from(components["java"]);
```
Comment (SgtSilvio) [inline build.gradle.kts:277]: This is already done in lines 208-228

File: build.gradle.kts [hunk 3]
```diff
@@ -269,46 +270,29 @@ allprojects {
...
-        // workaround for publishing gradle module metadata https://github.com/bintray/gradle-bintray-plugin/issues/229
-        tasks.withType<com.jfrog.bintray.gradle.tasks.BintrayUploadTask> {
-            doFirst {
-                publishing.publications.withType<MavenPublication> {
-                    val moduleFile = buildDir.resolve("publications/$name/module.json")
-                    if (moduleFile.exists()) {
-                        artifact(moduleFile).extension = "module"
-                    }
-                }
+            signing {
```
Comment (SgtSilvio) [inline build.gradle.kts:282]: why is this in afterEvaluate?

Comment (SgtSilvio) [reply build.gradle.kts:282]: I think `plugins.apply("signing")` is missing here

File: build.gradle.kts [hunk 4]
```diff
@@ -5,7 +5,8 @@ plugins {
     id("com.github.johnrengelman.shadow")
     id("biz.aQute.bnd.builder")
     id("maven-publish")
-    id("com.jfrog.bintray")
```
Comment (SgtSilvio) [inline build.gradle.kts:8]: please remove the `import java.util.*` in line 1, it was only used by the bintray plugin (as far as I can remember)

**Review by SgtSilvio — APPROVED** [review]

**Pull request merged**